bluepysnap/simulation.py

```diff
@@ -125,8 +136,17 @@ def simulator(self):
     @cached_property
     def node_sets(self):
         """Returns the NodeSets object bound to the simulation."""
-        path = self.to_libsonata.node_sets_file
-        return NodeSets.from_file(path) if path else NodeSets.from_dict({})
+        try:
+            path = self.circuit.to_libsonata.node_sets_path
+        except BluepySnapError:
+            path = ""
+
+        node_sets = NodeSets.from_file(path) if path else NodeSets.from_dict({})
```

Comment:
We could get a copy of `self.circuit.node_sets` instead, but it makes sense only if it makes the code simpler.

Reply:
Unless there's another use case for copying a node set, I'm not sure it's worth it.